Week3/homework/src/controllers/UsersController.js

```diff
@@ -0,0 +1,78 @@
+const IPublicOperations = require('./operations/IPublicOperations');
+const Queries = require('../database/Queries');
+const execQuery = require('../database');
+const faker = require('faker/locale/en');
+
+// Queries for USERS table
+const { userQueries, crudQueries } = new Queries('USERS');
+
+class UsersController extends IPublicOperations {
+  constructor(table) {
+    super(table);
+  }
+
+  // Custom query functions for USERS table
+  getUserTodos = async (req, res, next) => {
+    try {
+      const { id } = req.params;
+
+      const result = await execQuery(userQueries.selectUserTodos(), id);
+      res.json(result);
+    } catch (error) {
+      next(error);
+    }
+  };
+
+  getDetailedUsers = async (req, res, next) => {
+    try {
+      const { id } = req.params;
+      const details = await execQuery(userQueries.selectDetailedUsers());
+
+      let userId = [];
+
+      const result = details.reduce((arr, detail) => {
```

Comment:
Is it possible to have the same result by only using SQL queries?
> **GET** /api/users/2/details
```json
[
    {
        "id": 2,
        "name": "patrick",
        "surname": "anthropy",
        "email": "panthropy@mail.com",
        "todos": [
            {
                "title": "Eating",
                "description": "You neede to drink 2lt water everyday",
                "tags": "important, immediately",
                "is_done": 0,
                "deadline": "2019-08-11T20:13:10.000Z",
                "remind_date": "2019-08-12T20:10:10.000Z"
            },
            {
                "title": "Sport",
                "description": "Dont miss that match!",
                "tags": "not required",
                "is_done": 0,
                "deadline": "2019-08-11T20:13:10.000Z",
                "remind_date": "2019-08-12T20:10:10.000Z"
            },
            {
                "title": "Eating",
                "description": "You neede to drink 2lt water everyday",
                "tags": "important, immediately",
                "is_done": 0,
                "deadline": "2019-08-10T22:00:00.000Z",
                "remind_date": "2019-08-11T22:00:00.000Z"
            },
            {
                "title": "Sport",
                "description": "Dont miss that match!",
                "tags": "not required",
                "is_done": 0,
                "deadline": "2019-08-10T22:00:00.000Z",
                "remind_date": "2019-08-11T22:00:00.000Z"
            }
        ]
    }
]
```

Reply:
I am afraid that is not possible with sql queries. You would need to use at least two queries one to get the user and another to get the todos and then combine the data.